application/org.openjdk.jmc.ui/src/main/java/org/openjdk/jmc/ui/charts/XYChart.java

```diff
@@ -163,23 +164,31 @@ public IQuantity getSelectionEnd() {
 	}
 
 	public void renderChart(Graphics2D context, int width, int height) {
-		if (width > xOffset && height > Y_OFFSET) {
-			axisWidth = width;
+		if (width > xOffset && height > yOffset) {
+			axisWidth = width - xOffset;
 			// FIXME: xBucketRange and xTickRange should be more related, so that each tick is typically an integer number of buckets (or possibly 2.5 buckets).
 			xBucketRange = new SubdividedQuantityRange(currentStart, currentEnd, axisWidth, bucketWidth);
 			// FIXME: Use bucketWidth * ticksPerBucket instead of hardcoded value?
 			xTickRange = new SubdividedQuantityRange(currentStart, currentEnd, axisWidth, 100);
 			AffineTransform oldTransform = context.getTransform();
-			doRenderChart(context, height);
+			context.translate(xOffset, 0);
+			doRenderChart(context, height - yOffset);
 			context.setTransform(oldTransform);
 		}
 	}
 
+	public void renderTextCanvasText(Graphics2D context, int width) {
+		axisWidth = width;
+		AffineTransform oldTransform = context.getTransform();
+		doRenderTextCanvasText(context);
+		context.setTransform(oldTransform);
+	}
+
 	public void renderText(Graphics2D context, int width, int height) {
-		if (width >= xOffset && height > Y_OFFSET) {
-			axisWidth = width;
+		if (width > xOffset && height > yOffset) {
+			axisWidth = xOffset;
```

Comment:
Visually I can see why this change works, but there's one side-effect of making the `axisWidth` this size.

When a mouse wheel event occurs for a zoom in or out, the value of the mouse position is compared to the `axisWidth` to see if a zoom event should occur (at `XYChart.zoomXAxis()` line 434).
```
if ((x > 0) && (x < axisWidth)) { .. }
```
x is the x value of the mouse click. The problem here is that `axisWidth` would last be set to `xOffset`, which is 180. So the zoom-in/out on the regular charts will only work if the mouse wheel event happens within 180 of the start of the axis.

Reply:
I have addressed this issue in 370e3db